core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -705,25 +709,36 @@ class KafkaApis(val requestChannel: RequestChannel,
           metadataRequest.topics.asScala.toSet
       }
 
-    var (authorizedTopics, unauthorizedTopics) =
+    var (authorizedTopics, unauthorizedForDescribeTopics) =
       topics.partition(topic => authorize(request.session, Describe, new Resource(Topic, topic)))
 
+    var unauthorizedForCreateTopics = Set[String]()
+      
     if (authorizedTopics.nonEmpty) {
       val nonExistingTopics = metadataCache.getNonExistingTopics(authorizedTopics)
       if (config.autoCreateTopicsEnable && nonExistingTopics.nonEmpty) {
         authorizer.foreach { az =>
```

Comment:
Not part of this patch, but the `authorize` function seems to already handle this, so could we replace this and the `if` below with a call to `authorize(request.session, Create, Resource.ClusterResource)`?